tests/e2e_tests/config.py

```diff
@@ -0,0 +1,89 @@
+import base64
+import json
+import os
+import subprocess
+from pathlib import Path
+
+import boto3
+
+
+class EndToEndTestSecrets:
```

Comment:
nit: Messy class. It's probably easiest to write this way. But all these `if`'s in case it's local, could have been a class on its own.

Reply:
Fair call out.

See 2ade59187ef4236d13eeb2c74e50c7f2efef94a3 for updates - turned this into a protocol now 👀 Mild over-engineering, but a proof of concept for how we can do things without inheritance utilising mypy/type hinting.